### PR TITLE
Add class to all components and update types

### PR DESCRIPTION
**Pull request by hobbitronics**
## Preview of release

```[2:29:18 PM] [semantic-release] › ✔  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[2:29:18 PM] [semantic-release] › ℹ  The next release version is 10.1.0
[2:29:18 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[2:29:18 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
[2:29:18 PM] [semantic-release] › ⚠  Skip v10.1.0 tag creation in dry-run mode
[2:29:18 PM] [semantic-release] › ✔  Published release 10.1.0 on default channel
[2:29:18 PM] [semantic-release] › ℹ  Release note for version 10.1.0:
## 10.1.0 (https:/Users/mike/projects/ui-components//compare/v10.0.1...v10.1.0) (2023-03-29)
```
### Added

- DatatableCheckbox: Add class prop (1959e3d)
- DatatableCheckboxHeader: Add prop class (8618a4f)
- Divider: Add class prop (ae03e35)
- List.Item: Add class prop (bfd35c1)
- Subheader: Add class prop (af9990a)
- Switch: Add class prop (4ca247d)
- TabBar.Indicator: Add class prop (f60bf46)
- TabBar.Scroller: Add class prop (1f4e7d5)
- TabBar.Tab: add class prop (9ca980f)
- TopAppBar: Add class prop (077363e)

### Changed

- release-config: use conventionalcommits and custom release rules and notes with semantic release (3537856)
- types: update index.d.ts types (edd3f87)

### CODE REVIEW

File: components/mdc/Datatable/DatatableCheckbox.svelte [hunk 1]
```diff
@@ -9,7 +9,13 @@ onMount(() => dispatch('mounted'))
 const dispatch = createEventDispatcher()
 </script>
 
-<td class="mdc-data-table__cell mdc-data-table__cell--checkbox" on:click on:keypress on:keyup on:keydown>
+<td
+  class="mdc-data-table__cell mdc-data-table__cell--checkbox {$$props.class}"
```
Comment (Schparky) [inline components/mdc/Datatable/DatatableCheckbox.svelte:13]: Is this the kind of thing that @forevermatt was suggesting?
```suggestion
  class="mdc-data-table__cell mdc-data-table__cell--checkbox {$$props.class || ''}"
```

**Review by Schparky — APPROVED** [review]

**Comment by Schparky** [comment]
> ## Preview of release
> ```
> [2:29:18 PM] [semantic-release] › ℹ  The next release version is 10.1.0
> [2:29:18 PM] [semantic-release] › ℹ  Start step "generateNotes" of plugin "@semantic-release/release-notes-generator"
> [2:29:18 PM] [semantic-release] › ✔  Completed step "generateNotes" of plugin "@semantic-release/release-notes-generator"
> [2:29:18 PM] [semantic-release] › ⚠  Skip v10.1.0 tag creation in dry-run mode
> [2:29:18 PM] [semantic-release] › ✔  Published release 10.1.0 on default channel
> [2:29:18 PM] [semantic-release] › ℹ  Release note for version 10.1.0:
> ## 10.1.0 (https:/Users/mike/projects/ui-components//compare/v10.0.1...v10.1.0) (2023-03-29)
> 
> ### Added
> 
>     * DatatableCheckbox: Add class prop (1959e3d (https:/Users/mike/projects/ui-components//commit/1959e3d05792606a2c60a6d6e0a1f50d97ce3510))
>     * DatatableCheckboxHeader: Add prop class (8618a4f (https:/Users/mike/projects/ui-components//commit/8618a4f156965fd67febf61516d7c814ed01b149))
>     * Divider: Add class prop (ae03e35 (https:/Users/mike/projects/ui-components//commit/ae03e35f4c3c9314b9be4813bae28fcfba30494e))
>     * List.Item: Add class prop (bfd35c1 (https:/Users/mike/projects/ui-components//commit/bfd35c1e5c90df30aa9f84317795ff24ebe63863))
>     * Subheader: Add class prop (af9990a (https:/Users/mike/projects/ui-components//commit/af9990a46abeef3f884d1ae82ef2c5c6755b9a9c))
>     * Switch: Add class prop (4ca247d (https:/Users/mike/projects/ui-components//commit/4ca247dfc807d7ff4835b00a45ad4c6b92b533ca))
>     * TabBar.Indicator: Add class prop (f60bf46 (https:/Users/mike/projects/ui-components//commit/f60bf4628ddf52ed179e26c1e462f23e48995702))
>     * TabBar.Scroller: Add class prop (1f4e7d5 (https:/Users/mike/projects/ui-components//commit/1f4e7d527c772b5a3d25611ae0b04d7eac3a7168))
>     * TabBar.Tab: add class prop (9ca980f (https:/Users/mike/projects/ui-components//commit/9ca980fc4f0a6aad36563b34e74109a46b7347d1))
>     * TopAppBar: Add class prop (077363e (https:/Users/mike/projects/ui-components//commit/077363e18ac9b3c5ddd0577ed254a2df952da152))
> 
> ### Changed
> 
>     * release-config: use conventionalcommits and custom release rules and notes with semantic release (3537856 (https:/Users/mike/projects/ui-components//commit/3537856be3d599a89a6b054039c3965eb09b9c19))
>     * types: update index.d.ts types (edd3f87 (https:/Users/mike/projects/ui-components//commit/edd3f870f81da61a60bf9bc41b133a48cbd57abe))```
> ```

This description looks very machine-generated. My eye skips over it all because of the excess information.

File: components/mdc/Datatable/DatatableCheckbox.svelte (hunk 1 above)
Comment (hobbitronics) [reply components/mdc/Datatable/DatatableCheckbox.svelte:13]: yes it is, I was going to do them all in a seperate PR.

**Comment by hobbitronics** [comment]
@Schparky the release notes get better formatting when no in the terminal. Here's a screenshot of the release notes in the last release. 
image

**Comment by Schparky** [comment]
> @Schparky the release notes get better formatting when no in the terminal. Here's a screenshot of the release notes in the last release.

Yeah, that's better.

**Pull request merged**